actionpack/lib/action_controller/metal/redirecting.rb

```diff
@@ -50,6 +50,9 @@ module Redirecting
     #   redirect_to post_url(@post), status: 301, flash: { updated_post_id: @post.id }
     #   redirect_to({ action: 'atom' }, alert: "Something serious happened")
     #
+    # Statements after redirect_to in our controller get executed, so redirect_to doesn't stop the execution of the function.
+     # To terminate the execution of the function immediately after the redirect_to, use return.
```

Comment:
@simi : Good catch, will be fixed in https://github.com/rails/rails/pull/28296. :-)